[TarinPairor]ip

TarinPairor opened the pull request:
# Tartar Chatbot

Tartar is a simple chatbot designed to help you manage your tasks efficiently. With Tartar, you can easily add tasks to a list, mark or unmark them as completed, and even search for specific tasks.

> "Simplicity is the ultimate sophistication." - Leonardo da Vinci

## Tools Used

- [Java](https://www.java.com/en/)

## Features

- [ ] Add tasks to the list
- [ ] Mark tasks as completed
- [ ] Unmark tasks as completed
- [ ] List all tasks
- [ ] Search for specific tasks
- [ ] Delete tasks

## Usage

1. **Adding a Task**: To add a task, simply type `add <task>` in the chat window.
2. **Marking a Task**: To mark a task as completed, type `mark <task>` in the chat window.
3. **Unmarking a Task**: To unmark a completed task, type `unmark <task>` in the chat window.
4. **Listing Tasks**: To see all tasks, type `list` in the chat window.
5. **Searching for Tasks**: To find a specific task, type `find <keyword>` in the chat window.
6. **Deleting a Task**: To delete a task, type `delete <task>` in the chat window.

```
This is my main:
public static void main(String[] args) throws IOException {
                Duke d = new Duke();
                d.run();
            }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

TarinPairor committed:
Level-0

TarinPairor committed:
Added Ui and modified Duke

TarinPairor committed:
Added Task and TaskList

TarinPairor committed:
Add Process and isDone

TarinPairor committed:
Added event type

TarinPairor committed:
added simple expected and actual test case

TarinPairor committed:
implemented simple test cases

TarinPairor committed:
added delete and sloppy error handling

TarinPairor committed:
replaced map with enum in Task

TarinPairor committed:
added storing file with exception handling

TarinPairor committed:
added datetime formatting

TarinPairor committed:
added storage class

TarinPairor committed:
made packages for all classes but duke

TarinPairor committed:
Merge branch 'add-gradle-support'

TarinPairor committed:
added javadocs

TarinPairor committed:
fix build on gradle

TarinPairor committed:
add test cases for packages except processor

TarinPairor committed:
Create MANIFEST.MF

TarinPairor committed:
update javadoc details

TarinPairor committed:
Clean up to coding standard

TarinPairor committed:
Add method for finding task

TarinPairor committed:
Fix find command parsing error

TarinPairor committed:
Merge branch 'branch-A-CodingStandard'

TarinPairor committed:
Merge branch 'branch-Level-9'

TarinPairor committed:
Update Storage.java message

TarinPairor committed:
Storage: Update IOException error message

TarinPairor committed:
Remove comments

joelgoh1 commented on src/main/java/processor/Processor.java:
```diff
@@ -0,0 +1,221 @@
+     * @param userInput the user command to be processed
+     * @throws IOException if an I/O error occurs while processing the command
+     */
+    public void userInputProcessMarkUnmark(String userInput) throws IOException{
```
A space could be left between the exception and the curly braces, so that it aligns more accurately with Egyptian style.

TarinPairor replied:
I agree.

joelgoh1 commented on src/main/java/processor/Processor.java:
```diff
@@ -0,0 +1,221 @@
+     * @throws IOException if an I/O error occurs while processing the command
+     */
+    public void userInputProcessMarkUnmark(String userInput) throws IOException{
+        String[] array = userInput.split(" ");
```
Perhaps a more intuitive variable name here?

TarinPairor replied:
I agree that is very hard to read.

joelgoh1 commented on src/main/java/processor/Processor.java:
```diff
@@ -0,0 +1,221 @@
+        String substringAfterSpace = desc.substring(firstSpaceIndex + 1);
+
+        switch (command) {
+            case "todo":
```
There might be a need to reconfigure your IDE to have no indentations for switch cases to better align with coding standards.

TarinPairor replied:
Yep I agree. I resolved that later I believe.

joelgoh1 reviewed (commented):
LGTM! Just a few nits to fix!

TarinPairor committed:
Merge branch 'branch-Level-8'

TarinPairor committed:
Merge from 8 to master

TarinPairor committed:
Attempt at JavaFX/GUI in vain

TarinPairor committed:
Resolve error throwing in Duke

TarinPairor committed:
Merge branch 'branch-level-10'

TarinPairor committed:
Solve immediate Duke errors

TarinPairor committed:
Fix styling

TarinPairor committed:
Update README with appropriate markdown

TarinPairor committed:
Attempt to run JavaFX on Main/Launcher

TarinPairor committed:
Add ProcessorTest and a few assert keywords

TarinPairor committed:
Patch ProcessorTest failed cases

TarinPairor committed:
Successfully add abstraction of Processor

tayziyi commented on src/main/java/processor/Processor.java:
```diff
@@ -0,0 +1,268 @@
+    }
+
+
+    /**
```
Noted on nice way of writing Java doc to describe on a method. Noted this on other places as well

TarinPairor replied:
Appreciate it.

tayziyi commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,65 @@
+ * The Storage class is responsible for reading and writing tasks to the file.
+ */
+public class Storage {
+    private final String filePath;
```
Would be better if we use FILE_PATH here instead?

TarinPairor replied:
It would indeed I shall fix that.

tayziyi commented on src/main/java/tasks/Task.java:
```diff
@@ -0,0 +1,147 @@
+     * @param desc
+     * @param isDone
+     */
+    public Task(String desc, boolean isDone) {
```
Good choice on choosing way of describing your attributes in the constructor.

tayziyi commented on src/main/java/tasks/Task.java:
```diff
@@ -0,0 +1,147 @@
+     * Returns the type of the task.
+     * @return the type of the task
+     */
+    public TaskType getType() {
```
Noted on nice way in implementing getter and setter!

TarinPairor replied:
Thanks!

tayziyi commented on src/main/java/ui/Ui.java:
```diff
@@ -0,0 +1,134 @@
+
+public class Ui {
+
+    private static final String NAME = "Tartar";
```
Noted on significant use of caps in final constants! Good job

TarinPairor replied:
How nice of you.

tayziyi reviewed (commented):
LGTM! well done on following the guidelines on coding standard closely. Keep it up!

TarinPairor committed:
Add Javadocs to Factory

Factory.java: Currently does not have any JavaDocs.
It must have JavaDocs to follow coding conventions.

TarinPairor committed:
Make Processor variables more understandable

{current situation} -- use present tense

AddTaskProcessor.java: Variable names are not descriptive and there are no comments.
Descriptive comments must be added as well as modifications to the variable names.
This is to make it more descriptive of how the variables actually work.

TarinPairor committed:
Update Mark and Unmark Processors

MarkUnMarkTaskProcessor.java: Delete this and add 2 files to replace it.
This is to create a more consistent abstraction level for the mark and unmark functions.
MarkTask and UnmarkTask Processors have been created.

TarinPairor committed:
Utilize anyMatch stream method in AddTask

TarinPairor committed:
JavaFX works but can't use images

TarinPairor committed:
AddTaskProcessor.java: Add one assert method

TarinPairor committed:
Solve null pointer error with ImageViewer

TarinPairor committed:
Implement working GUI with fxml files and styles

TarinPairor committed:
Update styles.css file

TarinPairor committed:
Update non-functional jar file from duke.Launcher

TarinPairor committed:
Add functioning SortTaskProcessor.java

TarinPairor committed:
Implement interactive chatbot with format error

TarinPairor committed:
Add simple user guide

TarinPairor committed:
Add Ui.png

TarinPairor committed:
Follow checkstyle except some javadocs

TarinPairor committed:
Update javadocs for Processors

TarinPairor committed:
Add processor testcases

TarinPairor committed:
Add functional storage system

TarinPairor committed:
AddTaskProcessor.java: Improve abstraction quality

TarinPairor committed:
Add Launcher greeting message

TarinPairor committed:
Add thread in Duke.run()

TarinPairor committed:
Add pride personality and image

TarinPairor committed:
Update send button to be pink

TarinPairor committed:
Made ip more PG13

TarinPairor committed:
Fix README.md markdown bugs

TarinPairor committed:
Delete unnecessary QUOTE_STRING

TarinPairor committed:
Test Merging PRs

TarinPairor committed:
Merge pull request #2 from TarinPairor/branch-Merging-PRs

Test Merging PRs

TarinPairor committed:
Add javadocs to MainWindow.getResponse method

TarinPairor committed:
Merge branch 'master' of https://github.com/TarinPairor/ip
